// Protocol buffer for string to sequence encoding.
//
// Copyright 2019 the ProGraML authors.
//
// Contact Chris Cummins <chrisc.101@gmail.com>.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package ml4pl;

// An encoded string.
message EncodedString {
  repeated int32 encoded = 1;
}

// A StringEncoder job, containing a vocabulary, a list of strings to encode,
// and the encoded results.
message StringEncoderJob {
  map<string, int32> vocabulary = 1;
  repeated string string = 2;
  repeated EncodedString seq = 3;
}
